assertion_utils/index.ts

```diff
@@ -0,0 +1,39 @@
+
+export function forDataset(foo: string) {
+  return new DatasetAssertion(foo);
+}
+
+export class DatasetAssertion {
+  private readonly dataset: string;
+  private groupCols: string[] = [];
+  constructor(dataset: string) {
+    this.dataset = dataset;
+  }
+
+  public groupedBy(cols: string[]) {
```

Comment:
many of our APIs allow you to pass either a single string or a string array, i would suggest that we do the same here.

```
public groupedBy(cols: string | string[]) {
  this.groupCols = typeof cols === "string" ? [cols] : cols;
  return this;
}
```